core/events/events_block_move.ts

```diff
@@ -61,11 +61,25 @@ export class BlockMove extends BlockBase {
   newInputName?: string;
 
   /**
-   * The new X and Y workspace coordinates of the block if it is a top level
+   * The new X and Y workspace coordinates of the block if it is a top-level
    * block. Undefined if it is not a top level block.
    */
   newCoordinate?: Coordinate;
 
+  /**
+   * An explanation of what this move is for.  Known values include:
+   *  'drag' -- A drag operation completed.
+   *  'bump' -- Block got bumped away from an invalid connection.
+   *  'snap' -- Block got shifted to line up with the grid.
+   *  'inbounds' -- Block got pushed back into a non-scrolling workspace.
+   *  'connect' -- Block got connected to another block.
+   *  'disconnect' -- Block got disconnected from another block.
+   *  'create' -- Block created via XML.
+   *  'cleanup' -- Workspace aligned top-level blocks.
+   * Reasons may become comma separated if move events merge ('drag,bump,snap').
+   */
+  reason?: string;
```

Comment:
Can we make the list of reasons an enum? It would mean we get compile-time errors if we e.g. misspell `inbounds` or try to add a variation like `in-bounds`, plus give us autocomplete.

Also, can we have the reason property in the event be an array of reasons, instead of a single string with commas? That would be simpler for consumers of this property instead of having to parse the string into an array themselves.

Reply:
I don't think an enum would allow developers to create new reasons.  Imagine an extension or application which, say, shuffled blocks around randomly to create [a puzzle](https://blockly.games/puzzle).  They should be able to create a custom reason (e.g. 'shuffle').

Changed all reasons to be arrays.